pkg/controller/sparkapplication/controller.go

```diff
@@ -504,13 +510,14 @@ func (c *Controller) syncSparkApplication(key string) error {
 				// App will never be retried. Move to Terminal Failure State.
 				appToUpdate.Status.AppState.State = v1alpha1.FailedState
 				c.recordSparkApplicationEvent(appToUpdate)
-			} else if hasRetryIntervalPassed(appToUpdate.Spec.RestartPolicy.OnFailureRetryInterval, appToUpdate.Status.ExecutionAttempts, appToUpdate.Status.CompletionTime) {
+			} else if hasRetryIntervalPassed(appToUpdate.Spec.RestartPolicy.OnFailureRetryInterval, appToUpdate.Status.ExecutionAttempts, appToUpdate.Status.LastSubmissionAttemptTime) {
```

Comment:
I think it's correct to use `CompletionTime`, which is set if the driver pod phase is `Failed` or `Succeeded`. The application state is set to `FailingState` if the driver pod phase is `Failed`.

Reply:
If you kill the driver pod, `CompletionTime` wouldn't be set because it's set based on the completion time of the driver pod.

Reply:
Ok, so how to keep driver alive? It's particularly noticeable on short living nodes (eg GKE preemptible)

Reply:
This should be based on `CompetionTime` but to fix this we can potentially set CompetionTime while setting the state as `Failing` in https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/332/files#diff-b7eea120d7ac0e8587781b863b7db8e4R364 . We currently do metav1.Now() when we see driver pod phase is `Failed` or `Succeeded` . We will be overloading the term `CompletionTime` thou

Reply:
Ok. I'll do it that way.

Reply:
`TerminationTime` sounds good 👍